plugin/src/cli_utils/localpv/lvm/volume/mod.rs

```diff
@@ -0,0 +1,162 @@
+use kube::{api::ListParams, Api, Client};
+use prettytable::{row, Row};
+
+use lazy_static::lazy_static;
+
+lazy_static! {
+    pub(crate) static ref LVM_VOLUME_HEADER: Row =
+        row!["NAME", "NODE", "STATUS", "CAPACITY", "VOLGROUP", "PVC-NAME", "SC-NAME",];
+}
+
+/// Implementation for volumes cmd.
+pub(crate) async fn volumes(
+    cli_args: &CliArgs,
+    volumes_arg: &GetVolumesArg,
+    client: Client,
+) -> Result<(), Error> {
+    let volume_handle: Api<LvmVolume> = Api::namespaced(client.clone(), &cli_args.args.namespace);
+    let vol_list = lvm_volumes(volume_handle, volumes_arg)
+        .await
+        .map_err(|err| Error::Kube { source: err })?;
+    match get_lvm_vol_output(vol_list, &client).await {
+        Ok(lvol_record) => {
+            print_table(&cli_args.output, lvol_record);
+            Ok(())
+        }
+        Err(e) => Err(e),
+    }
+}
+
+/// Implementation for volume cmd.
+pub(crate) async fn volume(
+    cli_args: &CliArgs,
+    volume_arg: &GetVolumeArg,
+    client: Client,
+) -> Result<(), Error> {
+    let volume_handle: Api<LvmVolume> = Api::namespaced(client.clone(), &cli_args.args.namespace);
+    let volume = lvm_volume(volume_handle, volume_arg)
+        .await
+        .map_err(|err| Error::Kube { source: err })?;
+    match get_lvm_vol_output(vec![volume], &client).await {
+        Ok(lvol_record) => {
+            print_table(&cli_args.output, lvol_record);
+            Ok(())
+        }
+        Err(e) => Err(e),
+    }
+}
+
+/// Get lvm volume cr from the k8s cluster.
+async fn lvm_volume(
+    volume_handle: Api<LvmVolume>,
+    volume_arg: &GetVolumeArg,
+) -> Result<LvmVolume, kube::Error> {
+    let volume = volume_handle.get(&volume_arg.volume).await?;
+    Ok(volume)
+}
+
+/// Lists lvm volume cr from the k8s cluster. Retuns volumes from specific node if node is specified.
+async fn lvm_volumes(
+    volume_handle: Api<LvmVolume>,
+    volumes_arg: &GetVolumesArg,
+) -> Result<Vec<LvmVolume>, kube::Error> {
+    let max_entries = 500i32;
+    let mut lp: ListParams = if let Some(node_id) = &volumes_arg.node_id {
+        ListParams::default()
+            .labels(format!("kubernetes.io/nodename={}", node_id).as_str())
+            .limit(max_entries as u32)
+    } else {
+        ListParams::default().limit(max_entries as u32)
+    };
+    let mut vol_list = Vec::new();
+    loop {
+        let list = volume_handle.list(&lp).await?;
+        vol_list.extend(list.items);
+        match list.metadata.continue_ {
+            Some(token) if !token.is_empty() => {
+                lp = lp.continue_token(&token);
+            }
+            _ => break,
+        }
+    }
+    Ok(vol_list)
+}
+
+/// Converts Vec<LvmVolume> into LvolRecord.
+async fn get_lvm_vol_output(
+    lvm_vols: Vec<LvmVolume>,
+    client: &Client,
+) -> Result<LvmVolRecord, Error> {
+    let api: Api<PersistentVolume> = Api::<PersistentVolume>::all(client.clone());
+
+    let pvs = list_pv(api).await?;
```

Comment:
is there any label selector we can use to filter down the list?

Reply:
No, It doesnt have any labels at all.